blogposts/2020/announcing-sourcegraph-3.17.md

```diff
@@ -0,0 +1,258 @@
+
+Currently, a repository will need 50 navigation events (hovers, jump to definition, find references) to trigger the auto-indexing procedure. We are continually tuning these heuristics to bring precise results to more repositories faster.
+
+
+## Product preview: code insights
+
+TODO: Felix will make a video
+
+The Sourcegraph developer team is prototyping code insights, which will help you answer questions you have about your code, including: 
+
+   * [Code smells]([https://martinfowler.com/bliki/CodeSmell.html](https://martinfowler.com/bliki/CodeSmell.html)) over time
+   * Languages used at the organization
+   * Library adoption
+   * Test coverage over time
+   * Tracking a migration
+   * Security anti-patterns
+
+Code insights helps developers be more productive and do their jobs better, and brings the data we have in Sourcegraph together.
+
+We want to understand the bigger questions you have about your code base, and would love to hear from you! Please reach out at [feedback@sourcegraph.com](mailto:feedback@sourcegraph.com) to talk to us.
+
+
+## Sourcegraph developer features
+
+
+### AND/OR query improvements
+
+TODO: Rijnard is making a video
+
+In [Sourcegraph 3.15](https://about.sourcegraph.com/blog/sourcegraph-3.15#experimental-andor-operators-for-search-contents), we introduced AND and OR queries for file content for regexp and structural search modes as an experimental option. Sourcegraph 3.17 now introduces AND and OR queries for all modes (including the newly added literal search mode) and is on by default for all instances (no longer considered experimental).
+
+The AND/OR query improvements in this release only work for file content, but hierarchical AND/OR queries that include filters are on our product roadmap.
+
+
+### Gitolite exclude pattern
+
+The Gitolite `exclude` setting in the [Gitolite external service configuration](https://docs.sourcegraph.com/admin/external_service/gitolite#configuration) now supports a regular expression through the pattern field. Adding this regex support improves Gitolite’s pattern matching abilities.
+
+This is consistent with how we exclude in other external services, and is a replacement for the deprecated configuration. [#11403](https://github.com/sourcegraph/sourcegraph/pull/11403) 
+
+```
+
+TODO: Example config
+
+```
+
+<br />&nbsp;
+### Improved debugging experience
+
+A unified set of container metrics, compatible across both Kubernetes and Docker Compose deployment types,  provides the following information about containers through Grafana dashboards and alerting in both deployment types:
+
+ADD SCREENSHOTS - POSSIBLE TO STACK VERTICALLY TO IMPROVE READABILITY?
```

Comment:
@christinaforney do you have the original version of that screenshot? i can edit it to stack vertically

Reply:
I do not. I believe @slimsag or @bobheadxi added this screenshot.

Reply:
@slimsag @bobheadxi removing this placeholder text with a version of the screenshot. please send me the high-res version so i can update